x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_request/enrollment_api_keys.ts

```diff
@@ -14,12 +14,18 @@ import {
 
 type RequestOptions = Pick<Partial<UseRequestConfig>, 'pollIntervalMs'>;
 
-export function useGetOneEnrollmentAPIKey(keyId: string, options?: RequestOptions) {
-  return useRequest<GetOneEnrollmentAPIKeyResponse>({
-    method: 'get',
-    path: enrollmentAPIKeyRouteService.getInfoPath(keyId),
-    ...options,
-  });
+export function useGetOneEnrollmentAPIKey(keyId: string | undefined) {
+  return useConditionnalRequest<GetOneEnrollmentAPIKeyResponse>(
+    keyId
+      ? {
+          method: 'get',
+          path: enrollmentAPIKeyRouteService.getInfoPath(keyId),
+          shouldSendRequest: true,
```

Comment:
Could we drop the ternary and always send `shouldSendRequest: typeof keyId === "string"` or some other test?

Reply:
I can see what I can do here, if we do not have `keyId` we will not be able to build `path`  too

Reply:
Line 19 is evaluated in a Boolean-ish truthy/falsy value to determine which branch to execute. All I'm suggesting is using a single block and have `shouldSendResult` be the truthy/falsy value from line 19 or whatever the condition is.